Liquid filler fixes2

thebest9178 opened the pull request:
## Type of change

- [x] Bug fix
- [x] New feature

## Description

Fixed #709.
Made it possible to place more than one block at a time.
Set the default delay to 0.
Added max blocks per tick.
Added Shape and SortMode
Added ListMode for white and black lists

## Related issues

#709

# How Has This Been Tested?

By filling a river and small lava lake in creative with various settings.

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have added comments to my code in more complex areas.
- [x] I have tested the code in both development and production environments.

Wide-Cat commented:
![2023-08-20_18 58 01](https://github.com/MeteorDevelopment/meteor-client/assets/74723656/7818e404-d676-4e64-a1c0-87070d4f9ef0)
In this image, it's not placing any blocks, although it should be able to. My guess would be it's something to do with waterlogged blocks, because the same issue also appeared with a fence post under the water. 
I also still noticed a bit of weirdness with it not placing properly on sphere mode with the default range, but it's much less of an issue now, and I'm unsure what the root of that problem was.

thebest9178 commented:
Fixed the waterlogged blocks issue. It won't try to replace things it can't like kelp, but will replace stuff like sea grass.

I don't see any issues placing with default settings.
You mention it being less of an issue placing with sphere mode _now_, but before I added shapemode it only used a cube shape so there was no sphere mode before.
